### PR TITLE
refactor(iroh-net): rename PeerAddr to NodeAddr, introduce NodeId alias

### PR DESCRIPTION
**Pull request by divagant-martian**
## Description
First step on dealing with the whole peer_id vs node_id. Changes include:
- renames `PeerAddr` to `NodeAddr` and it's inner fields from `peer_id` to `node_id`
- introduces the `NodeId` type alias as stated in #1388 
- moves every field and parameter defined as `peer_id: PublicKey` to `node_id: NodeId`. As far as I can tell everywhere touched is not doing any crypto stuff.

## Notes & open questions

The one place where this is not changed is the tls code as it does deal with crypto ops. Later refactors should update the code to use `key: PublicKey` instead of `peer_id: PublicKey` for example.

In the future it might be worth to actually rename `PublicKey` to `NodeId` or creating a newtype `NodeId` that can only be obtained from a `PublicKey` but has no crypto ops available. Not doing any of that since I know it's controversial.

Also note that this is intentionally kept partial. Reasons for this:
- the state already has a mixed peer_id/node_id state, so we are not passing from a consistent state into an inconsistent one, but from a less desirable and inconsistent one to a more desirable, still inconsistent one
- keeping the PR short to facilitate a meaningful review

## Change checklist

- [x] Self-review.
- [ ] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

File: iroh-net/src/magic_endpoint.rs [hunk 1]
```diff
@@ -422,28 +422,28 @@ impl MagicEndpoint {
     /// If the `derp_region` is not `None` and the configured DERP servers do not include a DERP node from the given `derp_region`, it will error.
     ///
     /// If no UDP addresses and no DERP region is provided, it will error.
-    pub async fn connect(&self, peer_addr: PeerAddr, alpn: &[u8]) -> Result<quinn::Connection> {
-        self.add_peer_addr(peer_addr.clone())?;
+    pub async fn connect(&self, node_addr: NodeAddr, alpn: &[u8]) -> Result<quinn::Connection> {
+        self.add_peer_addr(node_addr.clone())?;
```
Comment (dignifiedquire) [inline iroh-net/src/magic_endpoint.rs:426]: should this method be renamed?

File: iroh-net/src/magicsock/peer_map.rs [hunk 2]
```diff
@@ -258,10 +258,10 @@ impl PeerMap {
 impl PeerMapInner {
     /// Get the known peer addresses stored in the map. Peers with empty addressing information are
     /// filtered out.
-    fn known_peer_addresses(&self) -> impl Iterator<Item = PeerAddr> + '_ {
+    fn known_peer_addresses(&self) -> impl Iterator<Item = NodeAddr> + '_ {
```
Comment (dignifiedquire) [inline iroh-net/src/magicsock/peer_map.rs:261]: rename method?

File: iroh-net/src/magicsock/peer_map.rs [hunk 3]
```diff
@@ -258,10 +258,10 @@ impl PeerMap {
 impl PeerMapInner {
```
Comment (dignifiedquire) [inline iroh-net/src/magicsock/peer_map.rs:258]: rename?

**Review by dignifiedquire — APPROVED** [review]
changes look good to me in general, what's the plan with things like `PeerMap`?

**Comment by Frando** [comment]
There's also still `MagicEndpoint::peer_id` I think.
What about a `s/peer_id/node_id` across the whole repo to also rename variables etc?

**Pull request merged**